src/nvim/os/env.c

```diff
@@ -410,14 +410,14 @@ static char *remove_tail(char *p, char *pend, char *name)
 /// @param[out] mustfree Ouput parameter for the caller to determine if they are
 ///           responsible for releasing memory. Must be initialized to false
 ///           by the caller.
-char_u *vim_getenv(char_u *name, bool *mustfree)
+char *vim_getenv(char *name, bool *mustfree)
 {
-  char_u      *p;
-  char_u      *pend;
+  char *p;
+  char *pend;
   int vimruntime;
 
 
-  p = (char_u *)os_getenv((char *)name);
+  p = (char *)os_getenv(name);
```

Comment:
Sorry, I don't agree to that. Purpose of this PR is, in fact, changing `char_u` into `char` at some places. But that doesn't mean the PR shouldn't contain anything else. It should, in fact, contain anything implied from the main purpose, to leave code as correct as it was before. Steps can be separated in different commits for clarity an ease of review, but the PR as a whole cannot introduce incorrect behaviour, or break previously correct things, not even temporarily. That would leave code in a worse state while your next PR gets merged.